### PR TITLE
Adds GH CI

### PR DESCRIPTION
**Pull request by jlaura**
## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
+      - name: Setup ENV
+        uses: conda-incubator/setup-miniconda@3b0f2504dd76ef23b6d31f291f4913fb60ab5ff3
+        with:
+          miniconda-version: "latest"
+          activate-environment: knoten
+          environment-file: environment.yml
+          auto-activate-base: false
+          auto-update-conda: true
+          python-version: ${{ matrix.python-version }}
```
Comment (acpaquette) [inline .github/workflows/ci.yml]: Might want to try the following here:
```
miniforge-version: latest
use-mamba: true
channels: conda-forge
activate-environment: ale
environment-file: environment.yml
auto-activate-base: false
auto-update-conda: true
python-version: ${{ matrix.python-version }}
```

Used this in ALE to fixed that somewhat random failing versions. Cobbled together from various issue posts across github

**Comment by jlaura** [comment]
@acpaquette This might be ready for review. CI passes for 3.8 and 3.9. ALE is failing to solve 3.10+. Is this expected?

**Comment by acpaquette** [comment]
It looks like ALE is succeeding on all versions of python on the main branch now. Recent previous main branch fails were due to unfixed tests in one of my PRs.

The potential fix I mentioned here was what I did in ALE to get the failing python versions working

**Comment by jlaura** [comment]
@acpaquette It looks like the released version of ALE is only python 3.8 and 3.9. That's what the solver thinks. This does not make sense though because I can see the releases [here](https://anaconda.org/conda-forge/ale/files) via condo-forge.

EDIT: I'm wrong, the issue isn't `ale`. I think it's `pysis`.

**Review by acpaquette — APPROVED** [review]

**Pull request merged**